base/src/main/java/org/aya/tyck/pat/PatTycker.java

```diff
@@ -288,20 +289,38 @@ private LhsResult checkLhs(Pattern.Clause match, Def.Signature signature) {
     return step1;
   }
 
+  /**
+   * @param outerPattern null if visiting the whole pattern (like `A, x, ctor a b`). This is only used for error reporting.
+   *                     For now, {@param outerPattern} is used when {@param sig} is not empty
+   *                     but {@param stream} is empty, it is possible when matching parameters of Ctor.
+   */
   public @NotNull Tuple2<SeqView<Pat>, Term>
-  visitPatterns(Def.Signature sig, SeqView<Pattern> stream) {
+  visitPatterns(@NotNull Def.Signature sig, @NotNull SeqView<Pattern> stream, @Nullable Pattern outerPattern) {
```

Comment:
In fact, I don't think that adding a new argument is good.

Reply:
It's ok